### PR TITLE
wip: added RemoteTable

### PR DESCRIPTION
**Pull request by mesejo**

### CODE REVIEW

File: python/letsql/common/utils/dask_normalize_expr.py [hunk 1]
```diff
@@ -55,7 +55,7 @@ def normalize_datafusion_databasetable(dt):
     if dt.source.name not in ("datafusion", "let"):
         raise ValueError
     ep_str = str(dt.source.con.table(dt.name).execution_plan())
-    if ep_str.startswith(("ParquetExec:", "CsvExec:")):
+    if ep_str.startswith(("ParquetExec:", "CsvExec:", "CustomExec")):
```
Comment (dlovell) [inline python/letsql/common/utils/dask_normalize_expr.py:58]: I think we want the process the remote expr, not the `ep_str` in this case

File: python/letsql/backends/let/__init__.py [hunk 2]
```diff
@@ -310,10 +343,12 @@ def fn(node, _, **kwargs):
     def _to_sqlglot(
         self, expr: ir.Expr, *, limit: str | None = None, params=None, **_: Any
     ):
-        op = expr.op()
-        out = op.map_clear(replace_cache_table)
+        expr = self._register_and_transform_cache_tables(expr)
```
Comment (dlovell) [inline python/letsql/backends/let/__init__.py:346]: why did we move this here?

File: python/letsql/backends/let/__init__.py [hunk 3]
```diff
@@ -290,8 +293,38 @@ def _get_source(self, expr: ir.Expr):
...
+                    if v in tables:
+                        name = f"{v.name}_{next(count)}"
+                        kwargs[k] = DatabaseTable(
+                            name,
+                            schema=v.schema,
+                            source=v.source,
+                            namespace=v.namespace,
+                        )
+                        remote: RemoteTable = tables[v]
+                        batches = remote.remote_expr.to_pyarrow_batches()
```
Comment (dlovell) [inline python/letsql/backends/let/__init__.py:311]: I think this doesn't address the case where there are multiple passes on the table within the single engine expr.

**Comment by mesejo** [comment]
Closed via #304

**Pull request closed without merging**